test/e2e/upg_e2e.go

```diff
@@ -604,6 +604,68 @@ var _ = ginkgo.Describe("CLI debug commands", func() {
 
 // TODO: validate both binapi and CLI against each other
 var _ = ginkgo.Describe("UPG Binary API", func() {
+	ginkgo.Context("for upf ueip pool", func() {
+		f := framework.NewDefaultFramework(framework.UPGModeTDF, framework.UPGIPModeV4)
+
+		addPool := func(isAdd bool, nwi string, name string) {
+			nwiConverted := append(util.EncodeFQDN(nwi), 0)
+			nameConverted := append(util.EncodeFQDN(name), 0)
+			req := &upf.UpfUeipPoolNwiAdd{
+				IsAdd:      isAdd,
+				Identity:   nwiConverted,
+				NwiName:    nameConverted,
+				NwiNameLen: uint8(len(nameConverted) - 1),
+			}
+			reply := &upf.UpfUeipPoolNwiAddReply{}
+
+			gomega.Expect(
+				f.VPP.ApiChannel.SendRequest(req).ReceiveReply(reply),
+			).To(gomega.Succeed(), "upf_tdf_ul_enable_disable")
+		}
+
+		type poolPairing struct {
+			nwi  string
+			name string
+		}
+
+		dumpPools := func() []poolPairing {
+			reqCtx := f.VPP.ApiChannel.SendMultiRequest(&upf.UpfUeipPoolDump{})
+			var ret []poolPairing
+			for {
+				msg := &upf.UpfUeipPoolDetails{}
+				stop, err := reqCtx.ReceiveReply(msg)
+				gomega.Expect(err).To(gomega.BeNil())
+				if stop {
+					break
+				}
+
+				identity := strings.TrimRight(string(msg.Identity), "\x00")
+				nwiName := strings.TrimRight(string(msg.NwiName), "\x00")
+
+				ret = append(ret, poolPairing{
+					nwi:  util.DecodeFQDN([]byte(identity)),
+					name: util.DecodeFQDN([]byte(nwiName)),
+				})
+			}
+
+			return ret
+		}
+
+		ginkgo.It("adds and removes a pool", func() {
+			addPool(true, "sgi", "mypool")
+			addPool(true, "test", "mypool")
+
+			pools := dumpPools()
+			gomega.Expect(pools).To(gomega.ContainElements(poolPairing{"sgi", "mypool"}, poolPairing{"test", "mypool"}))
+			gomega.Expect(len(pools)).To(gomega.Equal(2))
```

Comment:
If you use the suggestion below about `pool_foreach`, the elements will be in the order in which they were added, and you can just check equality of the slices

Reply:
I don't think the test should break if this order changes, so I'm for leaving it as is unless you insist to change it

Reply:
Ok, in this case you might rewrite it as 
```suggestion
			gomega.Expect(dumpPools()).To(gomega.ConsitOf(poolPairing{"sgi", "mypool"}, poolPairing{"test", "mypool"}))
```

See Gomega matcher reference here:
https://onsi.github.io/gomega/#provided-matchers